src/components/UserContribution.vue

```diff
@@ -0,0 +1,108 @@
+<script setup>
+import { ref, defineProps, computed } from "vue";
+import TextField from "./FieldTypes/TextField.vue";
+import MultipleSelect from "./FieldTypes/MultipleSelect.vue";
+import SingleSelect from "./FieldTypes/SingleSelect.vue";
+import { recordUserContribution } from "../stores/api.js";
+
+const dialog = ref(false);
+const data = defineProps({
+  description: String,
+  inputOptions: Object,
+  missingFieldName: String,
+  toolName: String,
+  taskId: Number,
+});
+
+const missingFieldValue = ref(null);
+const inputOptionsArray = computed(() => {
+  return data.inputOptions
+    ? Object.entries(data.inputOptions).map(([key, value]) => ({ key, value }))
+    : [];
+});
+
+async function postUserContribution() {
+  if (missingFieldValue.value.length > 0) {
+    const contributionRecord = {
+      value: Array.isArray(missingFieldValue.value) ? missingFieldValue.value.join() : missingFieldValue.value,
+      field: data.missingFieldName,
+      tool: data.toolName,
+    };
+
+    try {
+      await recordUserContribution(data.taskId, contributionRecord);
+    } catch (error) {
+      console.log(error);
+    } finally {
+      dialog.value = false;
+      missingFieldValue.value = null;
+    }
+  }
+}
+
+function closeDialog() {
+  dialog.value = false;
+  missingFieldValue.value = null;
+}
+
+</script>
+<template>
+  <v-dialog v-model="dialog" persistent>
+    <template v-slot:activator="{ props }">
+      <v-btn color="primary base100--text theme--light" v-bind="props">
+        Yes Please
+      </v-btn>
+    </template>
+    <v-container>
+      <v-card>
+        <v-card-title>
+          <span class="text-h5">Missing Field Form</span>
+        </v-card-title>
+        <v-card-text>
+          <p class="my-4">
+            Please help us add the missing information for below field
+          </p>
+
+          <v-row>
+            <TextField
+              v-if="!data.inputOptions"
+              v-model="missingFieldValue"
+              :missingFieldName="data.missingFieldName"
+              :description="data.description"
+            ></TextField>
+            <SingleSelect
+              v-if="
+                inputOptionsArray?.length > 0 &&
+                missingFieldName === 'tool_type'
+              "
+              v-model="missingFieldValue"
+              :missingFieldName="data.missingFieldName"
+              :inputOptions="inputOptionsArray"
+              :description="data.description"
+            ></SingleSelect>
+            <MultipleSelect
+              v-if="inputOptionsArray?.length > 0"
+              v-model="missingFieldValue"
+              :missingFieldName="data.missingFieldName"
+              :inputOptions="inputOptionsArray"
```

Comment:
The reason the input field is rending twice with a `tool_type` task is that both of these `v-if` statements are resolving to true.  

You could change the condition for MultipleSelect to 
```
            <MultipleSelect
              v-if="
                inputOptionsArray?.length > 0 &&
                missingFieldName !== 'tool_type'
              "
```